src/vmx.rs

```diff
@@ -756,6 +756,84 @@ pub fn read_flags() -> u64 {
     ret
 }
 
+#[naked]
+unsafe fn vmx_handle_vmexit() {
+    asm!(
+        "
+    push %rax
+    push %rbx
+    push %rcx
+    push %rdx
+    push %rbp
+    push %rsi
+    push %rdi
+    push %r8
+    push %r9
+    push %r10
+    push %r11
+    push %r12
+    push %r13
+    push %r14
+    push %r15
+    push %rsp
+
+    movq %rsp, %rdi
+
+    call vmx_dispatch
+
+    # Remove rsp from stack.
+    add $$8, %rsp
+
+    # Restore general purpose registers
+    pop %r15
+    pop %r14
+    pop %r13
+    pop %r12
+    pop %r11
+    pop %r10
+    pop %r9
+    pop %r8
+    pop %rdi
+    pop %rsi
+    pop %rbp
+    pop %rdx
+    pop %rcx
+    pop %rbx
+    pop %rax
+    vmresume
+
+    # We shouldn't get here.
+    movq $$0, %rax
+    setc %ah
+    setz %al
+    movq %rax, %rdi
+    jmp vmx_vmresume_failure
+    "
+    );
+}
+
+// Used by inline assembly.
+#[allow(dead_code)]
```

Comment:
Hiding stuff from the compiler concerns me. Since this function has a known ABI, couldn't we call it from vmx_handle_vmexit without using inline asm?